src/stactools/overture_maps/metadata.py

```diff
@@ -0,0 +1,446 @@
+                pyarrow_args = {
+                    "where": self.href,
+                    "filesystem": s3fs.S3FileSystem(
+                        **PYARROW_CONFIGS[self.storage_backend]
+                    ),
+                }
+            else:
+                pyarrow_args = {"where": self.href, "filesystem": None}
+
+            self._metadata = pq.read_metadata(**pyarrow_args)
+
+        return self._metadata
+
+    @property
+    def geo_metadata(self) -> Dict[str, Any]:
+        geo_metadata = json.loads(self.metadata.metadata[b"geo"].decode("utf-8"))
+        if not isinstance(geo_metadata, dict):
+            raise ValueError(f"could not parse geo metadata from {self.href}")
+
+        return geo_metadata
+
+    @property
+    def bbox(self) -> List[float]:
+        bbox: List[float] = self.geo_metadata["columns"]["geometry"]["bbox"]
+        if not all(isinstance(x, float) for x in bbox):
+            raise ValueError(
+                f"could not parse bounding box from {self.href}: {str(bbox)}"
+            )
+
+        return bbox
+
+    def to_item(self) -> Item:
+        collection = COLLECTION_ID_FORMAT.format(theme=self.theme)
+        asset = Asset(
+            href=self.href,
+            roles=["data"],
+            media_type=MediaType.PARQUET,
+        )
+
+        item = Item(
+            id=f"{self.theme}-{self.feature_type}-{self.release}-part-{self.part}-{self.uid}",
+            datetime=self._datetime,
+            bbox=self.bbox,
+            geometry=Polygon.from_bounds(*self.bbox).model_dump(exclude_none=True),
+            properties={
+                "overture:theme": self.theme,
+                "overture:type": self.feature_type,
+            },
+            assets={"data": asset},
+            collection=collection,
+        )
+
+        # version extension
+        VersionExtension.ext(item, add_if_missing=True)
+        item.ext.version.version = self.release
+
+        # table extension
+        table.TableExtension.ext(asset, add_if_missing=True)
+        if storage_options := PYARROW_CONFIGS.get(self.storage_backend):
+            asset.ext.table.storage_options = storage_options  # type: ignore
+
+        table.TableExtension.ext(item, add_if_missing=True)
+        item.ext.table.row_count = self.metadata.num_rows
+
+        item.ext.table.columns = [
+            table.Column(
+                properties={
+                    "name": col["name"],
+                    "col_type": col["type"]
+                    if isinstance(col["type"], str)
+                    else col["type"].get("type"),
+                }
+            )
+            for col in json.loads(
+                self.metadata.metadata[b"org.apache.spark.sql.parquet.row.metadata"]
+            )["fields"]
+        ]
+
+        # set primary geometry type to first geometry_type in geoparquet metadata
+        item.ext.table.primary_geometry = self.geo_metadata["columns"]["geometry"][
+            "geometry_types"
+        ][0]
```

Comment:
If there's a list of multiple geometry types, you really can't just pick one. It could be a mix of polygon and multi polygon and you'll lose that information